onedal/spmd/decomposition/incremental_pca.py

```diff
@@ -0,0 +1,114 @@
+# ==============================================================================
+# Copyright 2024 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+from daal4py.sklearn._utils import get_dtype
+
+from ...datatypes import _convert_to_supported, from_table, to_table
+from ...decomposition import IncrementalPCA as IncrementalPCA_nonSPMD
+from ...utils import _check_array
+from .._base import BaseEstimatorSPMD
+
+
+class IncrementalPCA(BaseEstimatorSPMD, IncrementalPCA_nonSPMD):
+    def _reset(self):
+        self._partial_result = super(IncrementalPCA_nonSPMD, self)._get_backend(
+            "decomposition", "dim_reduction", "partial_train_result"
+        )
+        if hasattr(self, "components_"):
+            del self.components_
+
+    def partial_fit(self, X, queue):
+        """Incremental fit with X. All of X is processed as a single batch.
+
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features)
+            Training data, where `n_samples` is the number of samples and
+            `n_features` is the number of features.
+
+        y : Ignored
+            Not used, present for API consistency by convention.
+
+        check_input : bool, default=True
+            Run check_array on X.
```

Comment:
Is it part of the API?

Reply:
yes, it is done like that in scikit

Reply:
Sorry, but I don't this check_input here in the method. Is this missed?

like here https://github.com/scikit-learn/scikit-learn/blob/70fdc843a/sklearn/decomposition/_incremental_pca.py#L252

Reply:
oh, sorry, it shouldn't be in the docstring, this parameter is used only in sklearnex part. fixed, thanks